src/main/java/redis/clients/jedis/BinaryJedis.java

```diff
@@ -2209,7 +2203,11 @@ public Long zcount(final byte[] key, final byte[] min, final byte[] max) {
   private Set<Tuple> getBinaryTupledSet() {
     checkIsInMulti();
     List<byte[]> membersWithScores = client.getBinaryMultiBulkReply();
-    Set<Tuple> set = new LinkedHashSet<Tuple>();
+    if (membersWithScores.size() == 0) {
+      return Collections.emptySet();
+    }
+    Set<Tuple> set = new LinkedHashSet<Tuple>(
+        CollectionUtils.mapCapacity(membersWithScores.size() / 2));
```

Comment:
@itugs now it's better. Go ahead and change the initialization to `new LinkedHashSet<Tuple>(membersWithScores.size() / 2, 1.0)`. We can remove the CollectionUtil class also